### PR TITLE
Use setter to prepend links with baseUrl

### PR DESCRIPTION
**Pull request by JoeChapman**
- Turn on middleware
- add custom middleware to prepend links with baseUrl

This is the first in a set of pull requests to remove the non-json responses. 
This handles the json response links

### CODE REVIEW

**Comment by chrisns** [comment]
does this work with socket requests?
last time I did stuff with like this that was compatible with socket requests I had to look at `req.headers || req.socket.handshake.headers`

File: config/http.js [hunk 1]
```diff
@@ -30,36 +32,62 @@ module.exports.http = {
...
+      '404',
+      '500',
+    ],
 
-  /****************************************************************************
-  *                                                                           *
-  * Example custom middleware; logs each request to the console.              *
-  *                                                                           *
-  ****************************************************************************/
+    setBaseUrl: (req, res, next) => {
```
Comment (chrisns) [inline config/http.js]: a config file doesn't feel like the right place for functional code, can this be moved to either a hook or maybe into `api/lib` ?

**Comment by JoeChapman** [comment]
@chrisns middleware only works with http(s) request as far as I can tell: http://sailsjs.org/documentation/concepts/middleware

File: config/http.js (hunk 1 above)
Comment (JoeChapman) [inline config/http.js]: how do you feel about `api/middleware` as a location?

**Pull request closed without merging**